backend/cmd/headlamp.go

```
@@ -1123,6 +1126,35 @@ func (c *HeadlampConfig) getClusters() []Cluster {
 	return clusters
 }
 
+func (c *HeadlampConfig) handleHeadlampInfo(context *kubeconfig.Context) error {
+	headlampInfo := context.KubeContext.Extensions["headlamp_info"]
+
+	// if the headlamp_info is not present we create it and keep the original cluster name
+	if headlampInfo == nil {
+		err := c.keepOriginalClusterName(context.SourceStr(), context.Name, false)
+		if err != nil {
+			logger.Log(logger.LevelError, map[string]string{"cluster": context.Name},
+				err, "failed to keep original cluster name")
+			return err
+		}
+	}
+
+	// this handles the clusters that have been renamed before the headlamp_info was introduced with original name
+	if headlampInfo != nil {
+		existingObj, marshalErr := MarshalCustomObject(headlampInfo, context.Name)
```

The marshalErr is silently ignored if there's an error. We should log and handle this error.

---

done!